vulnerabilities/scraper/rust.py

```diff
@@ -0,0 +1,90 @@
+import json
+import pytoml as toml
+from urllib.request import urlopen
+import urllib.request
+from urllib.error import HTTPError
+from io import BytesIO
+from zipfile import ZipFile
+from dephell_specifier import RangeSpecifier
+
+RUSTSEC_DB_URL = 'https://github.com/RustSec/advisory-db/archive/master.zip'
+
+
+def rust_crate_advisories(url, prefix='advisory-db-master/crates/'):
+    with urlopen(url) as response:
+        with ZipFile(BytesIO(response.read())) as zf:
+            for path in zf.namelist():
+                if path.startswith(prefix) and path.endswith('.toml'):
+                    yield toml.load(zf.open(path))
+
+
+def all_versions_of_crate(crate):
+    info_url = "https://crates.io//api/v1/crates/{}".format(crate)
+    info = json.load(urlopen(info_url))
```

Comment:
I can't find any mention in the documentation but I'm pretty sure the [`HTTPResponse`](https://docs.python.org/3/library/http.client.html#httpresponse-objects) object returned by `urlopen()` needs to be closed to avoid a resource leak. This is the reason for `with urlopen(url) as response:` in `rust_crate_advisories()`.